ledger/sandbox/src/test/suite/scala/com/digitalasset/platform/sandbox/services/command/CommandStaticTimeIT.scala

```diff
@@ -179,24 +246,39 @@ class CommandStaticTimeIT
       }
 
       "a choice on a contract is exercised before its LET" should {
-        "result in a rejection" in allFixtures { ctx =>
+        "result in a rejection" in {
           for {
-            commandClient <- ctx.commandClient()
-            getTimeResponse <- timeSource(ctx).runWith(Sink.head)
+            commandClient <- commandClient
+            getTimeResponse <- timeSource.runWith(Sink.head)
             currentTime = getTimeResponse.getCurrentTime
             templateId = templateIds.dummy
-            txEndOffset <- ctx.transactionClient.getLedgerEnd().map(_.getOffset)
-            comp <- commandClient.send(dummyCreateRequest(ctx, toInstant(currentTime), templateId))
-            _ = comp.getStatus should have('code(0))
-            transaction <- ctx.transactionClient.getTransactions(txEndOffset, None, TransactionFilter(Map(submitRequest.commands.value.party -> Filters.defaultInstance)))
+            txEndOffset <- TransactionServiceGrpc
+              .stub(channel)
+              .getLedgerEnd(GetLedgerEndRequest(unwrappedLedgerId))
+              .map(_.getOffset)
+            comp <- commandClient.send(dummyCreateRequest(toInstant(currentTime), templateId))
+            _ = comp.getStatus should have('code (0))
+            transaction <- ClientAdapter
+              .serverStreaming(
+                GetTransactionsRequest(
+                  unwrappedLedgerId,
+                  Some(txEndOffset),
+                  None,
+                  Some(TransactionFilter(
+                    Map(submitRequest.commands.value.party -> Filters.defaultInstance)))),
+                TransactionServiceGrpc
+                  .stub(channel)
+                  .getTransactions
+              )
+              .flatMapConcat(res => Source.fromIterator(() => res.transactions.iterator))
```

Comment:
Why not just `.flatMapConcat(Source(_.transactions))`?

Reply:
`Source` wants `scala.collection.immutable.Iterable[A]` and the value returned by the gRPC call is `scala.collection.Seq[A]` which is `<: scala.collection.Iterable[A]` but cannot be upcast to a `scala.collection.immutable.Iterable[A]`. I was a bit confused as well, but I can have a second look. :slightly_smiling_face:

Reply:
Addressed by https://github.com/digital-asset/daml/pull/3874/commits/ee05a8ccee3e4c282f52449d9536e91d23442ad0